### ISSUE TITLE
Add nfs-common packages to default steam os image (NAS breaks everytime it refreshes)

### ISSUE THREAD

**Issue by deadpahn** [ValveSoftware/SteamOS#963]
#### Your system information

* Steam client version: Latest
* SteamOS version: Latest
* Opted into Steam client beta?: [No] 
* Opted into SteamOS beta?: [No] 
* Have you checked for updates in Settings > System?: [Yes]

#### Please describe your issue in as much detail as possible:
Can you please add the nfs common files to the default steam image? I keep having to boot into desktop mode, unlock steam os, and download the files what seems to be every other week now. My deck is connected to my TV so its a pain to do when I wanna sit down and watch some TV via jellyfin/kodi when my food is ready.

#### Steps for reproducing this issue:

1. Not sure, but you need to install the nfs common library first.
2. Map your drive in fstab (make sure nofail is enabled as to not lock the deck when on the go/not connected to the network/internet)
Thanks

**Comment by ignacio82** [comment]
I just got my steam deck and one of the first things I wanted to do was to mount NFS shares from my NAS. I hope they add this soon.

**Comment by michaelBelsanti** [comment]
This would be really nice. Using NFS is my favorite way to move files to and from my deck, but having to reinstall it is inconvenient.

**Comment by Whitetigerswt** [comment]
Want to bump this. Please add this package to the default steam OS. In the meantime, I'm wondering if anyone has tried using systemd-sysext to layer the NFS package onto SteamOS to maintain the package after updates?

**Comment by Whitetigerswt** [comment]
I got tired of waiting around and I had some extra time today, so I went ahead and wrote a small bash script to do what I suggested in my last comment. Thanks to @bertogg for a blog post detailing how to do this.

```bash
#!/usr/bin/env bash

# https://blogs.igalia.com/berto/2022/09/13/adding-software-to-the-steam-deck-with-systemd-sysext/

mkdir working-dir && cd working-dir
wget "https://archive.archlinux.org/packages/n/nfs-utils/nfs-utils-2.6.2-1-x86_64.pkg.tar.zst"
mkdir nfs
tar -C nfs -xaf nfs-*.tar.zst usr

source /etc/os-release

mkdir -p "nfs/usr/lib/extension-release.d"
echo -e "ID=steamos\nVERSION_ID=${VERSION_ID}" > "nfs/usr/lib/extension-release.d/extension-release.nfs"

mksquashfs nfs nfs.raw

mkdir -p /var/lib/extensions
sudo mv nfs.raw /var/lib/extensions

sudo systemd-sysext refresh

sudo rm -r working-dir
```

In the script, note the version number  of `nfs-utils` the date of this comment. If there's a new version number of the `nfs-utils` from arch, you should install that one. https://archlinux.org/packages/core/x86_64/nfs-utils/

@bertogg What do you think about (working with me) and making a more generic package manager for SteamOS that installs packages through systemd-sysext? This would seem to allow a lot more Linux-like functionality on SteamOS.

**Comment by bertogg** [comment]
Hi @Whitetigerswt , I'm glad that it was useful.

I can see the utility of having a repository of systemd extensions for certain things but remember that extensions are not like distribution packages and they lack some very basic functionalities like dependencies, post-inst scripts and things like that (my blog post has more details). So I would not try to make anything sophisticated.

I'm happy to help with technical doubts but I don't really have the time for more.

**Comment by gravelfreeman** [comment]
> 

Could we have a repo with proper readme.md, description and installation instructions. Maybe we could update your bash script to add prompts to ask for the new values that may change like nfs-utils. Basically making it more user friendly.

Thanks for your work, I'll check it out eventually!

**Comment by gravelfreeman** [comment]
> FYI the current SteamOS 3.5 preview images come with `nfs-utils` preinstalled (I don't recommend them for regular use because they're untested and unstable).

Great news! It's been a month since your reply and was wondering if it has been tested so far.

I noticed I'm on version 3.4.10, when will version 3.5 expected to go stable?

Thank you!

**Comment by bertogg** [comment]
Yes, nfs-utils seems to work correctly on 3.5. I don't have the release date.

**Comment by gravelfreeman** [comment]
> Yes, nfs-utils seems to work correctly on 3.5. I don't have the release date.

Sorry where on Github or perhaps somewhere else can  I  see the current version?

**Comment by bertogg** [comment]
I'm not sure if I understand the question. The current version of the OS is in `/etc/os-release`, and for the `nfs-utils` package you can open a terminal and run `pacman -Q | grep nfs`
